File: app/schemas/sales.py
from __future__ import annotations
from pydantic import BaseModel, Field, ConfigDict
from typing import Optional
from decimal import Decimal
from datetime import date
from app.infra.models import SaleStatus
class SaleCreate(BaseModel):
    client_id: int
    user_id: int
    product_id: int

    total: Decimal = Field(gt=0)
    discount: Decimal = Field(default=Decimal("0.00"), ge=0)
    entry_amount: Optional[Decimal] = Field(default=None, ge=0)

    payment_type: str  # cash, pix, card e promissory

    installments_count: Optional[int] = Field(default=None, ge=1, le=60)
    first_due_date: Optional[date] = None

class SaleOut(BaseModel):
    model_config = ConfigDict(from_attributes=True)
    id: int
    public_id: str
    status: str
    payment_type: str
    total: Decimal
    discount: Decimal
    entry_amount: Optional[Decimal]

class SaleStatusUpdate(BaseModel):
    status: SaleStatus

File: wi-front/src/services/sales.ts
import { api } from "./api";

export type SaleCreate = {
  client_id: number;
  user_id: number;
  product_id: number;
  total: number;
  discount: number;
  entry_amount: number;
  payment_type: string;
  installments_count: number;
  first_due_date: string;
};

export async function createSale(payload: SaleCreate) {
  const { data } = await api.post("/sales", payload);
  return data;
}

export async function listSales() {
  const { data } = await api.get("/sales");
  return data;
}

export async function updateSaleStatus(saleId: number, status: string) {
  const { data } = await api.patch(`/sales/${saleId}/status`, { status });
  return data;
}
